### ISSUE TITLE
werkzeug: ... "OPTIONS / HTTP/1.0" 303

### ISSUE THREAD

**Issue by jbishop129** [odoo/docker#304]
I'm having trouble with Odoo v13 running as a stack. I noticed a HUGE amount of .sess session files building up over time in web-data/.local/share/Odoo/sessions, and they are generating about 1-2 per second. The web frontend container logs show a ton of:
`werkzeug: 10.0.0.2 - - [17/Mar/2020 01:31:49] "OPTIONS / HTTP/1.0" 303` 

I suspect it has something to do with the way the session passes thru Docker's networking layer. I have tried to see if the behavior is any different when accessing via a reverse-proxy, or directly to the Docker hosts' IP and container exposed port, same issue. This is a test deploy, zero user load on this.

Hoping I can get to the bottom of this; anyone seen this before? What other info can I provide to help troubleshoot?

**Comment by netfxtech** [comment]
I’m using Traefik (v2) as a proxy for Odoo v13. I only receive new session files when a user has logged into Odoo. There shouldn’t be any other reason a session is created. 

Deployed for 3 customers via normal docker-compose and then 2 other customers running this is in swarm. All record sessions the same way.

**Comment by codeagencybe** [comment]
@netfxtech 
Can I sent you PM? I have a question about your setup. 
I'm doing exactly the same with Traefik v2.2 and Docker swarm.

**Comment by netfxtech** [comment]
No problem. I probably won’t get to it for another 12-16hrs but I’ll respond when I receive it.

**Comment by codeagencybe** [comment]
@netfxtech 
I couldn't find a way to sent a PM here in Github, so I sent an email (I hope it was your company hahaha)

**Comment by drew-corporate-creations** [comment]
> I'm having trouble with Odoo v13 running as a stack. I noticed a HUGE amount of .sess session files building up over time in web-data/.local/share/Odoo/sessions, and they are generating about 1-2 per second. The web frontend container logs show a ton of: `werkzeug: 10.0.0.2 - - [17/Mar/2020 01:31:49] "OPTIONS / HTTP/1.0" 303`
> 
> [according to wikipedia ](https://en.wikipedia.org/wiki/HTTP_303)`303` has to do with web application redirection. I suspect it has something to do with the way the session passes thru Docker's networking layer; maybe it's causing some form of a redirect loop. I have tried to see if the behavior is any different when accessing via a reverse-proxy, or directly to the Docker hosts' IP and container exposed port, same issue. This is a test deploy, zero user load on this.
> 
> Hoping I can get to the bottom of this; anyone seen this before? What other info can I provide to help troubleshoot?

Did you ever figure out a solution for this? I have odoo deployed on AWS Fargate with the Data_Dir pointing to EFS. I noticed my EFS was increasing in size daily with nobody using it. Turns out, it's the same issue. Massive amounts of sessions building up. Any idea how to fix?

**Comment by codeagencybe** [comment]
Wild guess...perhaps it's something with your configuration and longpolling. 
There was a serious bug in a few versions with the longpolling and also fixed. 
Maybe you need to pull the latest docker image as well to make sure you have that fix..

**Comment by netfxtech** [comment]
I don't believe it has anything to do with your configuration. I manage 6 different odoo instances and I have 1 instance that does this. I cleared out the sessions folder yesterday and today I already have 1,526 sessions. The other instances have less than 20-30 sessions. These are very low usage Odoo's.

Now the only standout difference in the 1 odoo with 1,526 sessions is it's running 2 websites that are accessible from 2 different domains. Beyond that I don't know why it generates so many sessions.

**Comment by drew-corporate-creations** [comment]
Think I may have found a fix, though I haven't dug into the code to find out what it's actually doing.
My containers are behind a load balancer. There's a config option: --proxy-mode. I enabled that and it looks like the sessions are still getting created, but much slower. The session count on my filesystem has now started decreasing. So it seems like they are now getting created at a rate less than the rate that they expire.

**Comment by netfxtech** [comment]
My config
proxy_mode = True
workers = 4

Still, the sessions are rampant... I'm going to be deploying with Redis soon. If you ever figure it out please respond and let us know.
